marshmallow/schema.py

```diff
@@ -533,10 +529,10 @@ def dump(self, obj, many=None, update_fields=True, **kwargs):
                 self.__error_handler__(errors, obj)
             exc = ValidationError(
                 errors,
-                field_names=self._marshal.error_field_names,
-                fields=self._marshal.error_fields,
+                field_names=marshal.error_field_names,
+                fields=marshal.error_fields,
                 data=obj,
-                **self._marshal.error_kwargs
+                **marshal.error_kwargs
```

Comment:
This is the only field that the marshaller isn't passing to the `ValidationError`. The `ValidationError` constructor will accept kwargs and expose them for convenient access through a `kwargs` attribute.